### ISSUE TITLE
mod_full_brace_nl is inconsistent

### ISSUE THREAD

**Issue by pcercuei** [uncrustify/uncrustify#3181]
[//]: # "  If the issue is connected to existing Uncrustify options please, if possible, add the"
[//]: # "  following information to ease up the process:"
[//]: # "  • a link to a debug file:"
[//]: # "      generated with: 'uncrustify -p debug.txt -c pathToUsedConfig.cfg -f toBeFormatedFile.cpp' "
[//]: # "      Example: [debug.txt](https://linkToTheFile)"
[//]: # "      Example hosters for debug files: pastebin.com, gist.github.com, ..."
[//]: # "      The used config file is included in the debug file and does not need to be included here."
[//]: # 
[//]: # "  • include a small but complete test file that will be uncrustifyed"
[//]: # "  • include the generated results"
[//]: # "  • include the expected results"
[//]: # 
[//]: # "  ✋ please add a line containing ``` above and below of each of those three code sections"
[//]: # 
[//]: # "  • include the current version of your Uncrustify executable"
[//]: # "      printout via 'uncrustify -v'"
[//]: # "      Example: current version: uncrustify 0.63"
[//]: # "      or if possible additionally with the git sha of the commit"
[//]: # "                    current version: uncrustify 0.63 dc7b412"
[//]: # 
[//]: # "  • if possible include a version that worked"
[//]: # "      Example: working version: uncrustify 0.63"
[//]: # "      or"
[//]: # "                    working version: uncrustify 0.63 2a5e88f"

Using the following snipped of code as input:
```
for (i = 5; i < 10; i++)
	printf("%d\n",
	       i);

while (i < 15)
	printf("%d\n",
	       i);

if (i == 15)
	printf("%d\n",
	       i);
```

Using a configuration file that only has one line: `mod_full_brace_nl = 1`, the output is identical to the input.
[test1.txt](https://github.com/uncrustify/uncrustify/files/6500786/test1.txt)

Expected result:
```
for (i = 5; i < 10; i++) {
	printf("%d\n",
	       i);
}

while (i < 15) {
	printf("%d\n",
	       i);
}

if (i == 15) {
	printf("%d\n",
	       i);
}
```

Note that if I add `mod_full_brace_if_chain = true` to the script, Uncrustify will add braces around the `if`, but the `while` and `for` still won't get braces.

This is with uncrustify 54ab2de0.

**Comment by mwoehlke** [comment]
Are you looking for `mod_full_brace_{for,while}`?

**Comment by pcercuei** [comment]
@mwoehlke `mod_full_brace_{for,while}` would give me braces even on single lines like this:
```
for (i = 5; i < 10; i++) {
	printf("%d\n", i);
}
```
In the case where the statement spawns multiple lines, and in the case where the inner block spawns multiple lines, I do want braces. In the case where both the statement and inner block are single lines, I want to remove the braces.

**Comment by mwoehlke** [comment]
Yeah, I think this is a new feature request, then, and incidentally possibly the same as one I've come across recently (but hadn't yet attempted to file and/or implement). Namely:

```c++
  if (expr)
  { // Uncrustify should remove this
    statement;
  } // Uncrustify should remove this

  if (expr) // Uncrustify should add braces to this outer 'if'
    if (expr) // But not this inner one
      statement;
```

I suspect both our use cases would be implemented in a similar manner. Whether or not they'd be the same option could be debatable, though I'd expect a) your option enabled would give the behavior as above, and b) your option is actually what I want. (It's conceivable, however, that someone might want no braces for a single statement, even if it spans multiple lines, but want braces if the body is a compound statement.)

That said, do you want braces in this?
```c++
  if (expr)
    if (expr) statement;
```

If "no" or even "not necessarily", that would argue for having separate options, as neither is fully a subset of the other.

**Comment by pcercuei** [comment]
I would *think* you wouldn't want braces around your last example, but that's pretty twisted up - I'd never write something like that TBH.

**Comment by gmaurel** [comment]
You need to use the options:
```
mod_full_brace_if_chain = force
mod_full_brace_for      = force
mod_full_brace_if       = force
mod_full_brace_while    = force
```
